src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge39.java

```diff
@@ -0,0 +1,98 @@
+package org.owasp.wrongsecrets.challenges.docker;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+import java.util.Objects;
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import lombok.extern.slf4j.Slf4j;
+import org.owasp.wrongsecrets.RuntimeEnvironment;
+import org.owasp.wrongsecrets.ScoreCard;
+import org.owasp.wrongsecrets.challenges.Challenge;
+import org.owasp.wrongsecrets.challenges.ChallengeTechnology;
+import org.owasp.wrongsecrets.challenges.Difficulty;
+import org.owasp.wrongsecrets.challenges.Spoiler;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.annotation.Order;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Component;
+
+/** This is a challenge based on leaking secrets due to choice of filename as encryption key */
+@Slf4j
+@Component
+@Order(39)
+public class Challenge39 extends Challenge {
+
+  private final Resource resource;
+
+  public Challenge39(
+      ScoreCard scoreCard, @Value("classpath:executables/secrchallenge.md") Resource resource) {
+    super(scoreCard);
+    this.resource = resource;
+  }
+
+  @Override
+  public boolean canRunInCTFMode() {
+    return true;
+  }
+
+  @Override
+  public Spoiler spoiler() {
+    return new Spoiler(getSolution());
+  }
+
+  @Override
+  public boolean answerCorrect(String answer) {
+    return getSolution().equals(answer);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public int difficulty() {
+    return Difficulty.EASY;
+  }
+
+  /** {@inheritDoc} Cryptography based. */
+  @Override
+  public String getTech() {
+    return ChallengeTechnology.Tech.CRYPTOGRAPHY.id;
+  }
+
+  @Override
+  public boolean isLimitedWhenOnlineHosted() {
+    return false;
+  }
+
+  @Override
+  public List<RuntimeEnvironment.Environment> supportedRuntimeEnvironments() {
+    return List.of(RuntimeEnvironment.Environment.DOCKER);
+  }
+
+  private String getSolution() {
+    try {
+      String encryptedText = resource.getContentAsString(Charset.defaultCharset());
+      byte[] decodedEncryptedText = Base64.getDecoder().decode(encryptedText.trim());
+      byte[] decodedKey =
+          Objects.requireNonNull(resource.getFilename()).getBytes(StandardCharsets.UTF_8);
```

Comment:
https://github.com/OWASP/wrongsecrets/actions/runs/6427851204/job/17454371273?pr=1023 does not like the nullpointer derefernece introduced here. maybe start with getting the key and doing null checks on the resource?